utils/loading_cache.hh

```diff
@@ -446,15 +470,27 @@ private:
                 return true;
             }
             return false;
-        }, [this] (ts_value_lru_entry* p) {
+        };
+        auto value_destroyer = [this] (ts_value_lru_entry* p) {
             loading_cache::destroy_ts_value(p);
-        });
+        };
+
+        _new_gen_list.remove_and_dispose_if(expiration_cond, value_destroyer);
+        _lru_list.remove_and_dispose_if(expiration_cond, value_destroyer);
     }
 
-    // Shrink the cache to the _max_size discarding the least recently used items
+    // Shrink the cache to the _max_size discarding the least recently used items.
+    // Get rid from the entries that were used exactly once first.
```

Comment:
These two statements conflict, maybe need to reorder/rewrite them.

Reply:
Also, unless I'm misunderstanding the code as usual, the phrase "exactly once" seems to be true only for PartitionHitThreshold=1 - because this seems to remove everything in the "new generation", not only if its count is 1.

Reply:
> Also, unless I'm misunderstanding the code as usual, the phrase "exactly once" seems to be true only for PartitionHitThreshold=1 - because this seems to remove everything in the "new generation", not only if its count is 1.

You are correct (as usual ;)). When I wrote it I has a PartitionHitThreshold=1 in mind (I added this parameter later).

I'll rewrite the text. Thanks.